From 6089d5f4e9429cac74997c04e8cf8c60fedb1b15 Mon Sep 17 00:00:00 2001
From: Haozhong Zhang <hzzhan9@gmail.com>
Date: Sat, 17 May 2014 12:50:46 +0800
Subject: [PATCH] Fix gcc options to generate smaller bootloader.

---
 boot/Makefrag | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/boot/Makefrag b/boot/Makefrag
index bdb643b..1023633 100644
--- a/boot/Makefrag
+++ b/boot/Makefrag
@@ -16,16 +16,16 @@ BOOT_OBJS := $(OBJDIR)/boot/boot.o $(OBJDIR)/boot/main.o
 $(OBJDIR)/boot/%.o: boot/%.c
 	@echo + cc -Os $<
 	@mkdir -p $(@D)
-	$(V)$(CC) $(KERN_CFLAGS) -Os -c -o $@ $<
+	$(V)$(CC) $(KERN_CFLAGS) -fno-asynchronous-unwind-tables -Os -c -o $@ $<
 
 $(OBJDIR)/boot/%.o: boot/%.S
 	@echo + as $<
 	@mkdir -p $(@D)
-	$(V)$(CC) $(KERN_CFLAGS) -c -o $@ $<
+	$(V)$(CC) $(KERN_CFLAGS) -fno-asynchronous-unwind-tables -c -o $@ $<
 
 $(OBJDIR)/boot/main.o: boot/main.c
 	@echo + cc -Os $<
-	$(V)$(CC) $(KERN_CFLAGS) -Os -c -o $(OBJDIR)/boot/main.o boot/main.c
+	$(V)$(CC) $(KERN_CFLAGS) -fno-asynchronous-unwind-tables -Os -c -o $(OBJDIR)/boot/main.o boot/main.c
 
 $(OBJDIR)/boot/bootblock: $(BOOT_OBJS)
 	@echo + ld boot/bootblock
-- 
1.9.3